stl/inc/functional

```diff
@@ -29,6 +29,19 @@ _STL_DISABLE_CLANG_WARNINGS
 #undef new
 
 _STD_BEGIN
+#if _HAS_CXX23
+template <class _Result_type, class _Callable, class... _Types,
+    enable_if_t<is_invocable_r_v<_Result_type, _Callable, _Types...>, int> = 0>
+_NODISCARD constexpr _Result_type invoke_r(_Callable&& _Obj, _Types&&... _Args) noexcept(
+    is_nothrow_invocable_r_v<_Result_type, _Callable, _Types...>) {
+    if constexpr (is_void_v<_Result_type>) {
+        (void) _STD invoke(_STD forward<_Callable>(_Obj), _STD forward<_Types>(_Args)...);

```

Comment:
I observe that the underlying invoke machinery uses static_cast instead of forward for throughput, we might want to do that here, too

Reply:
Tracked by #1208 in general, but consistency is a good reason to do this now, I can validate and push changes.